compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -321,6 +321,28 @@ object Splicer {
     protected def interpretNew(fn: Symbol, args: => List[Result])(implicit env: Env): Result
     protected def unexpectedTree(tree: Tree)(implicit env: Env): Result
 
+    private final def removeEraisedArguments(args: List[Tree], fnTpe: Type): List[Tree] = {
+      var result = args
+      var index = 0
+      def loop(tp: Type): Unit = tp match {
```

Comment:
It's not the most beautiful code I've written but it's reasonable for the non erased path and (IMO) pretty clear

Reply:
I think we should modify `Call` to return a `List[List[Tree]]` for the arguments. It would make this logic much simpler as each element of the outer list corresponds to one of the method types.

Reply:
Indeed, it's much simpler! :)